src/iuse.cpp

```diff
@@ -6459,20 +6559,800 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+        const int radius, const tripoint bounds_center_point, const int bounds_radius,
+        const tripoint camera_pos, const units::volume &min_visible_volume, bool create_figure_desc,
+        std::unordered_set<tripoint> &ignored_points, std::unordered_set<std::string> &vehicles_recorded )
+{
+    const tripoint_range bounds = g->m.points_in_radius( bounds_center_point, bounds_radius );
+    const tripoint_range points_in_radius = g->m.points_in_radius( point, radius );
+    int dist = rl_dist( camera_pos, point );
+
+    bool item_found = false;
+    std::unordered_set<std::string> local_vehicles_recorded = std::unordered_set<std::string>
+            ( vehicles_recorded );
+    object_names_collection ret_obj;
+
+    std::string description_part_on_figure;
+    std::string description_furniture_on_figure;
+    std::string description_terrain_on_figure;
+
+    // store objects in radius
+    for( const tripoint &point_around_figure : points_in_radius ) {
+        if( !bounds.is_point_inside( point_around_figure ) ||
+            !g->m.sees( camera_pos, point_around_figure, dist + radius ) ||
+            ( ignored_points.find( point_around_figure ) != ignored_points.end() &&
+              !( point_around_figure == point && create_figure_desc ) ) ) {
+            continue; // disallow photos with not visible objects
+        }
+        units::volume volume_to_search = point_around_figure == bounds_center_point ? 0_ml :
+                                         min_visible_volume;
+
+        std::string furn_desc = colorized_feature_description_at( point_around_figure, item_found,
+                                volume_to_search );
+
+        const item item = get_top_item_at_point( point_around_figure, volume_to_search );
+        std::string item_name = colorized_item_name( item );
+
+        const optional_vpart_position veh_part_pos = g->m.veh_at( point_around_figure );
+        std::string unusual_ter_desc = colorized_ter_name_flags_at( point_around_figure,
+                                       camera_ter_whitelist_flags,
+                                       camera_ter_whitelist_types );
+        std::string ter_desc = colorized_ter_name_flags_at( point_around_figure );
+
+        const std::string trap_name = colorized_trap_name_at( point_around_figure );
+        const std::string field_desc = colorized_field_description_at( point_around_figure );
+
+        if( !furn_desc.empty() ) {
+            furn_desc = trap_name + furn_desc + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_furniture_on_figure = furn_desc;
+            } else {
+                ret_obj.furniture[ furn_desc ] ++;
+            }
+        } else if( veh_part_pos.has_value() ) {
+            const vehicle veh = veh_part_pos->vehicle();
+            const std::string veh_name = "<color_light_blue>" + veh.disp_name() + "</color>";
+            const std::string veh_hash = get_vehicle_hash( veh );
```

Comment:
Can't you just use the pointer to the vehicle instead of the hash? That hash does not seem to be serialized and the pointer is automatically unique.